storagemarket/impl/clientstates/client_fsm.go

```diff
@@ -209,10 +252,15 @@ var ClientEvents = fsm.Events{
 		From(storagemarket.StorageDealActive).To(storagemarket.StorageDealError).
 		Action(func(deal *storagemarket.ClientDeal, err error) error {
 			deal.Message = xerrors.Errorf("error waiting for deal completion: %w", err).Error()
+			deal.AddLog(deal.Message)
 			return nil
 		}),
 	fsm.Event(storagemarket.ClientEventFailed).
-		From(storagemarket.StorageDealFailing).To(storagemarket.StorageDealError),
+		From(storagemarket.StorageDealFailing).To(storagemarket.StorageDealError).
+		Action(func(deal *storagemarket.ClientDeal) error {
+			deal.AddLog(deal.Message)
```

Comment:
Is `deal.Message` set here?

Reply:
I don't think so, however we definitely want to call `AddLog` so that we update the timestamps for the stage. Changing to "".